permits/tests.py

```diff
@@ -212,6 +213,33 @@ def test_submit_permit_request_sends_email_to_secretariat(self):
         self.assertEqual(len(emails), 1)
         self.assertEqual(emails[0].to, ["secretariat@yverdon.ch"])
 
+    def test_missing_mandatory_date_property_gives_invalid_feedback(self):
+        permit_request = factories.PermitRequestFactory(author=self.user.permitauthor)
+        factories.WorksObjectTypeChoiceFactory(permit_request=permit_request)
+        permit_request.administrative_entity.works_object_types.set(
+            permit_request.works_object_types.all()
+        )
+        prop = factories.WorksObjectPropertyFactory(
+            input_type=models.WorksObjectProperty.INPUT_TYPE_DATE, is_mandatory=True
+        )
+        prop.works_object_types.set(permit_request.works_object_types.all())
+
+        data = {
+            "properties-{}_{}".format(works_object_type.pk, prop.pk): ""
+            for works_object_type in permit_request.works_object_types.all()
+        }
+
+        response = self.client.post(
+            reverse(
+                "permits:permit_request_properties",
+                kwargs={"permit_request_id": permit_request.pk},
+            ),
+            data=data,
+        )
+        parser = get_parser(response.content)
+        parser.select(".invalid-feedback")
+        self.assertEqual(1, len(parser.select(".invalid-feedback")))
```

Comment:
Les assertions se font plutôt dans l’autre sens (`actual, expected`).

Reply:
Correct and corrected! :)